### DISCUSSION TITLE
mod to move only the start button to the left of the screen?

### DISCUSSION THREAD

**Discussion by slrgt** [ramensoftware/windhawk-mods#679] [Q&A] [answered]
Hi, I love having my apps centered on the taskbar but I would like to move only the start button to the left of the screen since I almost never use it. Is there a mod that does this? I can't seem to find one

**Comment by m417z** [comment]
There's no mod which does exactly that, but it should be possible with the [Windows 11 Taskbar Styler](https://windhawk.net/mods/windows-11-taskbar-styler) mod.
You can find various examples here:
https://github.com/ramensoftware/windows-11-taskbar-styling-guide/blob/main/README.md#themes

UWPSpy is helpful for exploring which styles can be applied for which elements.

**Reply by slrgt** [reply]
I tried using that but i couldn't get it to move the start button to the left on its own :/

**Reply by m417z** [reply]
Try taking the target from here:
https://github.com/ramensoftware/windows-11-taskbar-styling-guide/blob/main/README.md#hide-the-start-button

But instead of hiding, you want to move. Try using the style from here:
https://github.com/ramensoftware/windows-11-taskbar-styling-guide/issues/12#issuecomment-1906680866

Of course, adjust the numbers as needed.

**Reply by slrgt** [reply]
Thanks that worked! The only issue is that since opening new apps moves everything in the taskbar to the left if i open two apps the start menu dissapears. I wish there was a way to have it Align Left instead of just moving it X number of pixels.

Here's what the settings look like if anyone else is curious (the main difference to the link provided is changing the transform to X and then adjusting the number based on your screen size.
![VSCodium_6WSJPByJo3](https://github.com/ramensoftware/windhawk-mods/assets/114287507/ea6c08bc-9a97-4a50-b02c-6944e2a97a2b)

if you want to copy paste 

Target:
`Taskbar.ExperienceToggleButton#LaunchListButton[AutomationProperties.AutomationId=StartButton]`
Styles:
`Transform3D:=<CompositeTransform3D TranslateX="-725" />`

**Reply by m417z** [reply]
I think it's better to align the whole taskbar to the left (see my comment in the other question), and then, if you want, add a margin on the start button's right side to move the taskbar buttons to the right.

**Reply by slrgt** [reply, marked as answer]
that works perfectly! for anyone curious align your taskbar to the left by right clicking it, then left click taskbar settings, taskbar behaviors, and set Taskbar Alignment to Left. Then in the settings of Windhawk Windows 11 Taskbar Styler add this
Target:
`Taskbar.ExperienceToggleButton#LaunchListButton[AutomationProperties.AutomationId=StartButton]`
Styles:
`Margin=0,0,625,0` 
(adjust the number depending on your screen size, just change it and click save until the pinned apps are centered)
![VSCodium_Idj4IF0ISh](https://github.com/ramensoftware/windhawk-mods/assets/114287507/928dd8c5-6a0e-4c8b-80d8-8836cb5a2e5a)

**Reply by PinkSerenity** [reply]
This is still not the perfect solution since it aligns the apps to the left and simply adds a margin. So adding more apps will make them fill the remaining space to the right.
<img width="1044" alt="image" src="https://github.com/ramensoftware/windhawk-mods/assets/41524896/8aaad053-6179-4130-b2f8-57e1d4c6f787">
Additionally it moves the widgets button to the center as well. To add a margin to the right of the widgets button, use `Taskbar.AugmentedEntryPointButton#AugmentedEntryPointButton[AutomationProperties.AutomationId=WidgetsButton]` as a target instead.

**Reply by LeviPumarlo** [reply]
--> "This is still not the perfect solution since it aligns the apps to the left and simply adds a margin." 
Agreed

**Reply by PinkSerenity** [reply]
One might analyze how to make a custom [Layout](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.layout) for the taskbar. I already tried defining a new one in Taskbar-Styler, but that only made the UI continously restart.

Edit: Reading into it more, it looks kinda easy, at least the XAML part. Writing the mod to apply the layout is another thing.

**Reply by peterjay733** [reply]
Tried moving start button to left with Windhawk while using Open Shell. Unfortunately the Open Shell start button now covers the leftmost icon on the centered taskbar. Not a bug in Windhawk, just not the way "we" would like it. Unfortunately Open Shell used to put the Open Shell icon on left even though taskbar was centered, but an Open Shell update "fixed" that. I liked it the way it was.

If you're using Windhawk to do this without something like Open Shell, the mod seems so be just fine.
